### ISSUE TITLE
would using lurch/rooted remove the need for red-green deployment?

### ISSUE THREAD

**Issue by thegostep** [spalladino/ethereum-upgrade-storage-free#1]
https://github.com/ricmoo/lurch/tree/master/rooted

**Comment by spalladino** [comment]
Not really. Root has the same restriction as any other CREATE2-based upgrade, which is having to deploy the upgraded version in a different tx than the SELFDESTRUCT.

**Comment by thegostep** [comment]
and using a factory to bundle the calls doesn't get around this restriction? I find this surprising

**Comment by spalladino** [comment]
As far as I understand, problem is that SELFDESTRUCT does not destroy the contract - it only marks it for destruction, and the destruction is carried out when the tx finishes. I ran some tests locally and it seems to be the case.
